File: testdata/java/level-08-abstract-inheritance/src/Circle.java
public class Circle extends Shape {
    private double radius;

    public Circle(double radius) {
        super("red");
        this.radius = radius;
    }

    @Override
    public double area() {
        return Math.PI * radius * radius;
    }

    public double circumference() {
        return 2 * Math.PI * radius;
    }
}

File: testdata/php/level-06-inheritance/src/circle.php
<?php

require_once 'base.php';

class Circle extends Shape {
    private float $radius;

    public function __construct(float $radius) {
        $this->radius = $radius;
    }

    public function area(): float {
        return 3.14159 * $this->radius * $this->radius;
    }
}

function main(): void {
    $c = new Circle(5.0);
    echo $c->area() . "\n";
    echo $c->describe() . "\n";
}
